prov/efa/test/efa_unit_test_info.c

```diff
@@ -151,3 +152,75 @@ void test_info_check_shm_info()
 	test_info_check_shm_info_from_hints(hints);
 
 }
+
+/**
+ * @brief Check that cases when a user requested FI_HMEM support
+ *        using different version of libfabric API, the EFA provider
+ *        behavior as expected on a system has CUDA
+ */
+void test_info_check_hmem_cuda_support_on_api_version()
+{
+	struct fi_info *hints, *info = NULL;
+	int err;
+
+	if (!hmem_ops[FI_HMEM_CUDA].initialized)
+		return;
+
+	hints = efa_unit_test_alloc_hints(FI_EP_RDM);
+
+	hints->caps |= FI_HMEM;
+	hints->domain_attr->mr_mode |= FI_MR_HMEM;
+
+	/* When a user used >= 1.18 API, CUDA is always supported.
+	 * This is because in version 1.18, libfabric EFA provider implemented
+	 * universal CUDA support through CUDA library. However, this feature
+	 * can cause some middleware to deadlock, thus it is only available
+	 * when a user is using 1.18 API.
+	 */
+	err = fi_getinfo(FI_VERSION(1, 18), NULL, NULL, 0, hints, &info);
+	assert_int_equal(err, 0);
+	fi_freeinfo(info);
+
+	/* now, check the behavior when user is using older version of API */
+
+	if (efa_device_support_rdma_read()) {
+		/* on a system that support GPUDirect RDMA read,
+		 * HMEM cuda is available when GPUDirect RDMA is available,
+		 * and environment variable FI_EFA_USE_DEVICE_RDMA set to 1/on/true
+		 */
+		setenv("FI_EFA_USE_DEVICE_RDMA", "1", true /* overwrite */);
+		err = fi_getinfo(FI_VERSION(1, 14), NULL, NULL, 0, hints, &info);
+		assert_int_equal(err, 0);
+		fi_freeinfo(info);
```

Comment:
Any reason not to separate this into another test, i.e. `test_xxx_on_api_lt_1_18` and `test_xxx_on_api_ge_1_18`

Reply:
I will do that.